### ISSUE TITLE
Need simple example of checking if a FIle Exists on Windows in a PCL Analyzer VSIX

### ISSUE THREAD

**Issue by paul1956** [dsplaisted/PCLStorage#29]
I think I need the code below
```
           Dim f As IFolder = Await FileSystem.Current.LocalStorage.GetFolderAsync(IO.Path.GetDirectoryName(context.Document.FilePath), context.CancellationToken)
            Dim r As ExistenceCheckResult = Await f.CheckExistsAsync(newFileName, context.CancellationToken)
            If r = ExistenceCheckResult.FileExists Then
 ```
but when I run it in the debugger I get this message on the first line
```
"This functionality is not implemented in the portable version of this assembly.  You should reference the PCLStorage NuGet package from your main application project in order to reference the platform-specific implementation."
```
What does this mean and how do I fix it? I don't need this in the VSIX project because it is not a PCL and the PCL project can't access the VSIX project.

**Comment by dsplaisted** [comment]
PCL Storage uses the ["bait and switch"](http://log.paulbetts.org/the-bait-and-switch-pcl-trick/) pattern, so you do need to reference PCL storage from the VSIX even if you're not using it directly there.

**Comment by paul1956** [comment]
Sorry I am being dense but I think you are saying to get this to work all I have to do is add the PCLStorage NuGet package to the VSIX project in addition to the PCL project. Then when I run the code above it will just work. If that is correct I will close the issue.

**Comment by dsplaisted** [comment]
Yes, try referencing the PCLStorage NuGet package from the VSIX project and let me know if that works.  If not, perhaps VSIX projects work a little differently and you might have to manually include the right DLL.

**Comment by paul1956** [comment]
That did not work, the PCL project reference
```
...\packages\PCLStorage.1.0.2\lib\portable-net45+wp8+wpa81+win8+monoandroid+monotouch+Xamarin.iOS+Xamarin.Mac\PCLStorage.dll
```
![image](https://cloud.githubusercontent.com/assets/12789369/11922064/b50eb07e-a747-11e5-95b3-ce88dcf1907b.png)
The copy in the VSIX is different.
```
...\packages\PCLStorage.1.0.2\lib\net45\PCLStorage.dll
```
![image](https://cloud.githubusercontent.com/assets/12789369/11922097/34917f0c-a748-11e5-9a46-6763701ee93f.png)

**Comment by dsplaisted** [comment]
The references are supposed to be different. That's how bait and switch works.

-----Original Message-----

**Comment by paul1956** [comment]
But it doesn't work, no switching is going on.
